onadata/libs/filters.py

```diff
@@ -355,6 +364,13 @@ def _xform_filter(self, request, view, keyword):
             self.xform = get_object_or_404(XForm, pk=xform)
             xform_qs = XForm.objects.filter(pk=self.xform.pk)
             public_forms = XForm.objects.filter(pk=self.xform.pk, shared_data=True)
+        elif filename:
+            attachment_id = view.kwargs.get("pk")
+            attachment = get_object_or_404(Attachment, pk=attachment_id)
+            xform = attachment.instance.xform.pk
+            self.xform = get_object_or_404(XForm, pk=xform)
```

Comment:
Why can't we do `self.xform = attachment.instance.xform` and avoid getting the xform again?